### PR TITLE
Update S3 tutorial and related docs

### PR DESCRIPTION
**Pull request by rdettai**
### Description

While reading the S3 tutorial I noticed some inconsistencies that made me want to run it again. 
- the ingestion was crashing on t4g.medium (even after tweaking the `commit_timeout_secs` config), so increasing the suggested config to t4g.large.
- section on time pruning was removed as this dataset does not present clear benefits from using it. To balance that removal, I added a mention to it in the REST API docs.

### How was this PR tested?

Running the tutorial step by step on EC2 instances.

### CODE REVIEW

**Comment by rdettai** [comment]
Note: I am not addressing https://github.com/quickwit-oss/quickwit/issues/3939 because there are a few missing things:
- we should test whether it still works if we public dataset bucket is in a different regions as the user
- the dataset used `hdfs-logs-multitenants.json.gz` is zipped which isn't supported by the file source. Note that hosting the uncompressed version might become a bit costly when users download it from outside AWS or another region (0.15 per download)

File: docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md [hunk 1]
```diff
@@ -244,49 +271,6 @@ Now that you have a search cluster, ideally, you will want to load balance exter
 This can quickly be done by adding an AWS load balancer to listen to incoming HTTP or HTTPS traffic and forward it to a target group.
 You can now play with your cluster, kill processes randomly, add/remove new instances, and keep calm.
 
-
```
Comment (fmassot) [inline docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md:247]: Can you open an issue to later add a better example on time-based pruning?

File: docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md [hunk 2]
```diff
@@ -165,7 +166,33 @@ You can check it's working by using `search` subcommand and look for `ERROR` in
...
+      "tenant_id": 51,
+      "timestamp": 1469687755
+    },
+    ...
+  ],
+  "elapsed_time_micros": 522542
+}
+```
+
+You can see that this query has 345 hits and that the server responds in 523 milliseconds.
```
Comment (fmassot) [inline docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md:193]: If you got `523 milliseconds` with a cold start, it's nice to add this information. Once the metastore is loaded, I would expect a lower response time.

File: docs/reference/cli.md [hunk 3]
```diff
@@ -336,7 +336,7 @@ quickwit index ingest
 | `--batch-size-limit` | Size limit of each submitted document batch. |
 | `--wait` | Wait for all documents to be commited and available for search before exiting |
 | `--force` | Force a commit after the last document is sent, and wait for all documents to be committed and available for search before exiting |
-| `--commit-timeout` | Duration of the commit timeout operation. |
+| `--commit-timeout` | Timeout for ingest operations that require waiting for the final commit (`--wait` or `--force`). This is different from the `commit_timeout_secs` indexing setting which sets the maximum time before commiting splits after their creation. |
```
Comment (fmassot) [inline docs/reference/cli.md:339]: nice

**Review by fmassot — APPROVED** [review]

File: docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md (hunk 1 above)
Comment (rdettai) [reply docs/get-started/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md:247]: https://github.com/quickwit-oss/quickwit/issues/3985

**Comment by rdettai** [comment]
While running the bench again I noticed that the tutorial was failing. As the problem I noticed probably also applies to other docs, I opened an issue instead of addressing it here: https://github.com/quickwit-oss/quickwit/issues/3984

**Pull request merged**